android/app/src/main/AndroidManifest.xml

```diff
@@ -13,61 +13,96 @@
-    <uses-feature android:name="android.hardware.touchscreen" android:required="false" />
+    <uses-permission
+        android:name="android.permission.READ_PHONE_STATE"
+        tools:node="remove" />
+    <uses-permission
+        android:name="android.permission.CAMERA"
+        android:required="false" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+
+    <uses-feature
+        android:name="android.hardware.touchscreen"
+        android:required="false" />
 
     <application
-      android:name=".MainApplication"
-      android:allowBackup="true"
-      android:label="@string/app_name"
-      android:icon="@mipmap/ic_launcher"
-      android:theme="@style/AppTheme">
-      <activity
-        android:name=".MainActivity"
+        android:name=".MainApplication"
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
-        android:windowSoftInputMode="adjustResize"
-        android:launchMode="singleTask">
-        <intent-filter>
-            <action android:name="android.intent.action.MAIN" />
-            <category android:name="android.intent.category.LAUNCHER" />
-        </intent-filter>
-        <intent-filter>
-            <action android:name="android.intent.action.VIEW" />
-            <category android:name="android.intent.category.DEFAULT" />
-            <category android:name="android.intent.category.BROWSABLE" />
-            <data android:scheme="zulip" android:host="login" />
-        </intent-filter>
-      </activity>
+        android:theme="@style/AppTheme">
+        <activity
+            android:name=".MainActivity"
+            android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
+            android:label="@string/app_name"
+            android:launchMode="singleTask"
+            android:windowSoftInputMode="adjustResize">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:host="login"
+                    android:scheme="zulip" />
+            </intent-filter>
+        </activity>
+
+        <activity
+            android:name=".sharing.ReceiveShareActivity"
+            android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
+            android:label="@string/app_name"
+            android:launchMode="singleTask"
+            android:windowSoftInputMode="adjustResize">
```

Comment:
>Are these extra attributes helpful and do they do things we want?
`windowSoftInputMode` and `configChanges` are not relevant here, because we don't care how the activity responds to the keyboard etc. But we do need `launchMode` to be `singleTask` - we want only one instance of the activity to exist at any given time.

I'll remove the useless attribs.

Reference: https://developer.android.com/guide/topics/manifest/activity-element

Reply:
>Hmm, in fact. I wonder if it might simplify things to simply add these intent filters to the main activity. After all, the behavior we're actually implementing here amounts to that in the end -- the user goes to send something, i.e. causes an intent with action android.intent.action.SEND, and what we end up doing is passing that all to the main activity.

One possible problem with that would be that we always have to launch an activity. So, in case the app is already running, we will definitely launch MainActivity for every share action. We can close it soon, but I would expect the `ZulipMobile` component to be pretty heavy in terms of resources.

Reply:
> We always have to launch an activity.

Do we? I think this is what `android:launchMode` controls, and we can make a different choice.

In fact, I think the value `singleTask` already means it won't launch a new activity. Here's [the documentation for that attribute](https://developer.android.com/guide/topics/manifest/activity-element#lmode), and here's what it says about `singleTask`:
> The system creates the activity at the root of a new task and routes the intent to it. However, if an instance of the activity already exists, the system routes the intent to existing instance through a call to its onNewIntent() method, rather than creating a new one.

(I definitely agree we don't want to create a new MainActivity when we already have one running.)